virttest/utils_test/qemu.py

```diff
@@ -1842,3 +1843,301 @@ def get_all_memorys(self, vm):
         session = self.get_session(vm)
         output = session.cmd_output_safe(cmd, timeout=90)
         return set([_ for _ in output.splitlines() if _])
+
+
+class UsbDevTest(object):
+
+    """
+    Provide basic test for usb devices
+    """
+
+    def __init__(self, test, params, env):
+        self.params = params
+        self.test = test
+        self.env = env
+        self.vm = env.get_vm(params["main_vm"])
+
+    def check_usb_dev(self, usb_type, info_usb_name,
+                      vp_id, vendor, product):
+
+        """
+        check if the usb device exist in guest
+        """
+        if usb_type == 'usb-storage' or usb_type == 'usb-bot':
+            self.add_udisk()
+        timeout = float(self.params.get("login_timeout", 600))
+        session = self.vm.wait_for_login(timeout=timeout)
```

Comment:
Please call vm.verify_alive() to check is VM really alive before login, so that you can release tester from trouble in find login failed reason.